community/consistency-check/src/main/java/org/neo4j/consistency/CheckConsistencyCommand.java

```diff
@@ -0,0 +1,168 @@
+import org.neo4j.logging.LogProvider;
+import org.neo4j.server.configuration.ConfigLoader;
+
+import static org.neo4j.dbms.DatabaseManagementSystemSettings.database_path;
+
+public class CheckConsistencyCommand implements AdminCommand
+{
+    public static class Provider extends AdminCommand.Provider
+    {
+        public Provider()
+        {
+            super( "check-consistency" );
+        }
+
+        @Override
+        public Optional<String> arguments()
+        {
+            return Optional.of( "--database=<database> [--tuning-config=<file>] [--verbose]" );
+        }
+
+        @Override
+        public String description()
+        {
+            return "Check the consistency of a database.";
+        }
+
+        @Override
+        public AdminCommand create( Path homeDir, Path configDir, OutsideWorld outsideWorld )
+        {
+            return new CheckConsistencyCommand( homeDir, configDir );
+        }
+    }
+
+    private final Path homeDir;
+    private final Path configDir;
+    private final ConsistencyCheckService consistencyCheckService;
+    private final LogProvider logProvider;
+    private final FileSystemAbstraction fileSystemAbstraction;
+
+    public CheckConsistencyCommand( Path homeDir, Path configDir )
+    {
+        this( homeDir, configDir, new ConsistencyCheckService() );
+    }
+
+    public CheckConsistencyCommand( Path homeDir, Path configDir, ConsistencyCheckService consistencyCheckService )
+    {
+        this.homeDir = homeDir;
+        this.configDir = configDir;
+        this.consistencyCheckService = consistencyCheckService;
+        this.logProvider = FormattedLogProvider.toOutputStream( System.out );
+        this.fileSystemAbstraction = new DefaultFileSystemAbstraction();
+    }
+
+    @Override
+    public void execute( String[] args ) throws IncorrectUsage, CommandFailed
+    {
+        String database;
+        Boolean verbose;
+        File tuningConfigFile;
+
+        Args parsedArgs = Args.parse( args );
+        try
+        {
+            database = parsedArgs.interpretOption( "database", Converters.mandatory(), s -> s );
+            verbose = parsedArgs.getBoolean( "verbose" );
+            tuningConfigFile =
+                    parsedArgs.interpretOption( "tuning-confing", Converters.optional(), Converters.toFile() );
+        }
+        catch ( IllegalArgumentException e )
+        {
+            throw new IncorrectUsage( e.getMessage() );
+        }
+
+        Config config = loadNeo4jConfig( homeDir, configDir, database, loadTuningConfig( tuningConfigFile ) );
+
+        try
+        {
+            consistencyCheckService.runFullConsistencyCheck( config.get( database_path ), config,
+                    ProgressMonitorFactory.textual( System.err ), logProvider, fileSystemAbstraction, verbose );
+        }
+        catch ( ConsistencyCheckIncompleteException | IOException e )
+        {
+            throw new CommandFailed( "Consistency checking failed.", e );
```

Comment:
This is not very helpful. I think you at least need to include the exception message.